runtime/include/chpl-mem.h

```diff
@@ -211,6 +224,24 @@ void* chpl_mem_layerAlloc(size_t, int32_t lineno, int32_t filename);
 void* chpl_mem_layerRealloc(void*, size_t, int32_t lineno, int32_t filename);
 void chpl_mem_layerFree(void*, int32_t lineno, int32_t filename);
 
+//
+// Does the implementation provide allocated memory that is already
+// localized to the calling sublocale?  That is, when the locale model
+// has sublocales and an allocation is done while running on one, will
+// the allocated memory be localized to that sublocale?  (Note that the
+// answer doesn't have to be completely truthful; it really only matters
+// for allocations large enough that we'll try to force localization
+// where chpl_mem_doLocalization is true, above.)
+//
+static inline
+bool chpl_mem_allocLocalizes(void) {
+#ifdef CHPL_MEM_IMPL_ALLOCLOCALIZES
+    return CHPL_MEM_IMPL_ALLOCLOCALIZES;
```

Comment:
Not a bit, and this may also change depending on where we end up with the optional-implementation question.